### PR TITLE
Split delegation and default chat agent

### PR DESCRIPTION
**Pull request by planger**
Also remove the special case with the delegation agent label and truly support multiple chat agents chiming in.

Fixes https://github.com/eclipsesource/osweek-2024/issues/125

PS: I think we should rename the Chat Agents...
1. *DefaultChatAgent* isn't a good name, we should choose something more meaningful, like *Software Dev Expert*
2. All agents shouldn't have agent in their name imho.
3. *DelegatingChatAgent* could be called *Dispatcher* or so.

PPS: I didn't optimize the prompt template and agent descriptions for the delegation yet, they didn't seem to work very well today.

### CODE REVIEW

File: packages/ai-chat/src/common/delegating-chat-agent.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                ...message,
+                request: {
+                    ...message.request,
+                    displayText: `@${agentId} ${message.request.text}`,
+                    isHidden: true
+                }
+            }, agentId);
+            await this.chatAgentService.invokeAgent(agentId, newRequest);
+        }
+    }
```
Comment (sdirix) [inline packages/ai-chat/src/common/delegating-chat-agent.ts:119]: This seems more of a hack to me: We pollute the chat models with "user" requests which never happened. For the UI to not look bad, we then hide those additional requests. However to anyone inspecting the ChatModel it looks like the same request was dispatched multiple times.
Also the way the framework is currently set up, these additional requests are by default sent to all LLMs invoked later.

I think we should either:
 - Extend the ChatModel/RequestModel to support multiple responses per request, or
 - Extend the ResponseParts to handle multiple responses of different sources

Comment (planger) [reply packages/ai-chat/src/common/delegating-chat-agent.ts:119]: Hm, yeah I see. You are right that this is a bit ugly.

I wanted to embrace the metaphor of conversation participants (agents) and therefore wanted to explicitly mention the delegation in the chat model, but there is no easy way to represent a request that isn't sent manually by the user. So another approach could be to enable agents to also post requests and not just responses. Could be interesting in the light of the trendy "mixture of agents" approach.

But I don't have a hard opinion. I also like your suggestion to support multiple responses per request.

**Comment by sdirix** [comment]
@planger I tried my luck with simplifying the delegation (in a separate commit) but I'm not sure I like the result. Please have a look.

**Comment by planger** [comment]
I like the approach, thank you!

Yes, on a second consideration, probably it is a bit verbose to show a delegation message for each delegation and following the metaphor of a multi-agent conversation is good and fine, but the delegation probably doesn't add too much value to that.

**Pull request merged**